### PR TITLE
cleanup: fix golint errors in /pkg/kubelet/stats

### PR DESCRIPTION
**Pull request by AndersonQ**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix golint failures in /pkg/kubelet/stats

**Which issue(s) this PR fixes**:

Part of #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1143,8 +1143,8 @@ type Kubelet struct {
 	// It should be set only when docker is using non json-file logging driver.
 	dockerLegacyService legacy.DockerLegacyService
 
-	// StatsProvider provides the node and the container stats.
-	*stats.StatsProvider
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/kubelet.go:1147]: I'd suggest to use explicit name here and keep it as `StatsProvider`. Having it like this `klet.Provider` is very confusing

**Comment by SergeyKanzhelev** [comment]
Build error (please build locally for validation):

```
ERROR: missing input file 'pkg/kubelet/stats/stats_provider.go', owner: '//pkg/kubelet/stats:stats_provider.go'
ERROR: /home/prow/go/src/github.com/kubernetes/kubernetes/pkg/kubelet/stats/BUILD:63:1: //pkg/kubelet/stats:go_default_test: missing input file '//pkg/kubelet/stats:stats_provider.go'
ERROR: /home/prow/go/src/github.com/kubernetes/kubernetes/pkg/kubelet/sysctl/BUILD:23:1: Creating runfiles tree bazel-out/k8-fastbuild/bin/pkg/kubelet/sysctl/linux_amd64_stripped/go_default_test.runfiles failed due to build-runfiles failed: error executing command 
```

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1143,8 +1144,8 @@ type Kubelet struct {
...
+	// Provider provides the node and the container stats.
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/kubelet.go:1147]: this must have cause linter errors as name in comment doesn't match the name of a field:

```suggestion
	// StatsProvider provides the node and the container stats.
```

Comment (SergeyKanzhelev) [reply pkg/kubelet/kubelet.go:1147]: @AndersonQ please address this comment

File: pkg/kubelet/stats/BUILD [hunk 3]
```diff
@@ -3,13 +3,13 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 go_library(
     name = "go_default_library",
     srcs = [
-        "cadvisor_stats_provider.go",
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/stats/BUILD:6]: why this file name change? Type is still called cadvisorStatsProvider so no need to change the name of the file.

**Comment by SergeyKanzhelev** [comment]
@AndersonQ can you please fix this last comment:

image

Everything else looks good. Thank you!

**Comment by SergeyKanzhelev** [comment]
lgtm

/retest
/assign @dashpole 
/assign @dims

File: pkg/kubelet/kubelet.go [hunk 4]
```diff
@@ -1135,6 +1136,56 @@ type Kubelet struct {
...
+func (kl *Kubelet) GetContainerInfo(podFullName string, uid types.UID, containerName string, req *cadvisorapi.ContainerInfoRequest) (*cadvisorapi.ContainerInfo, error) {
+	return kl.StatsProvider.GetContainerInfo(podFullName, uid, containerName, req)
+}
+
+// GetRawContainerInfo check pkg/kubelet/stats.Provider interface
+func (kl *Kubelet) GetRawContainerInfo(containerName string, req *cadvisorapi.ContainerInfoRequest, subcontainers bool) (map[string]*cadvisorapi.ContainerInfo, error) {
+	return kl.StatsProvider.GetRawContainerInfo(containerName, req, subcontainers)
+}
+
+// RlimitStats check pkg/kubelet/stats.Provider interface
```
Comment (MHBauer) [inline pkg/kubelet/kubelet.go:1184]: are these comments indicating that I should go read the other interface? 

Could be something like "delegates to an embedded instance of the stats.Provider interface".

**Review by MHBauer — APPROVED** [review]
/lgtm 

seems fine if I am interpreting the comment correctly.

File: pkg/kubelet/kubelet.go (hunk 4 above)
Comment (AndersonQ) [reply pkg/kubelet/kubelet.go]: @MHBauer I improved the doc comment

**Comment by riking** [comment]
You're still looking for an approver here.

**Comment by SergeyKanzhelev** [comment]
/assign @dims

for the hack/OWNERS

**Comment by SergeyKanzhelev** [comment]
@dims can you pls approve for hack/owners?

/assign @dims

**Comment by aojea** [comment]
my 2 cents, if all jobs fails it usually means you have something wrong :smile: 
In this case seems there is an undefined variable
```
/b/f/w/pkg/kubelet/kubelet.go:1150:38: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1155:50: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1160:63: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1165:37: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1170:74: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1175:86: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1180:36: undefined: statsapi
/b/f/w/pkg/kubelet/kubelet.go:1195:36: undefined: statsapi
```

/uncc

**Comment by AndersonQ** [comment]
@dchen1107, the PR is fixed :)

**Pull request merged**